### PR TITLE
ThemesShowcase: reduce searching flickering.

### LINKED ISSUE #12274: Themes: blinking content between searches
cap-

While using search on themes, the content keeps blinking the screen to the loader placeholder, and that makes for a fairly negative experience as everything seems moving around for no reason.

* Currently: list with thumb search 1 → blinking loaders → list with thumb search 2
* Ideally:  list with thumb search 1 → list with thumb search 2

Additionally the PR should add an `is-loading` class to the search component to allow prototyping alternatives to show the loading time.

### PR DESCRIPTION
**Pull request by budzanowski**
### Info
This fixes #12274

WORK IN PROGRESS.

### CODE REVIEW

**Comment by folletto** [comment]
Currently live:
cap-current


This PR changes:
cap-

It seems working well! Way way less load flickering, and no skeleton loaders in the list while typing.

It's still doing the skeleton on the title, but apart from that it's a solid improvement.

**Comment by folletto** [comment]
Ok so my more detailed feedback would be:

1. Awesome work in general as it drastically reduces the flickering
2. Would be ideal to remove also the occasional "no themes found" glitch, but as we discussed it's a larger refactoring so it's not up to this PR
3. We should avoid if possible to show the skeleton loader for the ”WordPress·com themes (x)" title too, and just update the number instead.
4. We shouldn't show the odd loader icon in place of the lens, it actually adds more visual change than it's useful. I'd rather just add a class `is-busy`to the whole search component if possible, to be styled later. I guess this might be a different component level change (Search?) so we can discuss and address in a separate PR.

**Comment by folletto** [comment]
I noticed a regression: if I click `×` it loses focus.

File: client/my-sites/themes/theme-showcase.jsx [hunk 1]
```diff
@@ -201,9 +203,14 @@ const ThemeShowcase = React.createClass( {
...
 
 export default connect(
-	( state, { siteId } ) => ( {
-		isLoggedIn: !! getCurrentUserId( state ),
-		siteSlug: getSiteSlug( state, siteId ),
-		isJetpack: isJetpackSite( state, siteId ),
-	} )
+	( state, { siteId } ) => {
+		const isRequestingSite = hasSiteActiveQueryRequests( state, siteId );
+		const isRequestingWpcom = hasSiteActiveQueryRequests( state, 'wpcom' );
```
Comment (ockham) [inline client/my-sites/themes/theme-showcase.jsx]: Can't we use any of these [existing](https://github.com/Automattic/wp-calypso/blob/8c4345bc104b40c510526bb1894aed2157cc3de0/client/state/themes/selectors.js#L150L162) [selectors](https://github.com/Automattic/wp-calypso/blob/8c4345bc104b40c510526bb1894aed2157cc3de0/client/state/themes/selectors.js#L255-L285)?

**Comment by folletto** [comment]
What's the status on this one?

**Comment by budzanowski** [comment]
If #12562 will be approved this will not be used.

**Comment by budzanowski** [comment]
Closed in favor of #12562 - which was also closed but had a valid working code that was just a bit too controversial. The future fix will probably be #12562 spinoff and not this one.

**Pull request closed without merging**